UsbVirtualPort.cpp

```diff
@@ -0,0 +1,195 @@
+        printf("Baudrate for the write port = %d\n", dcbSerialParamsW.BaudRate);
+        printf("ByteSize for the write port = %d\n", dcbSerialParamsW.ByteSize);
+        printf("StopBits for the write port = %d\n", dcbSerialParamsW.StopBits);
+        printf("Parity for the write port   = %d\n\n", dcbSerialParamsW.Parity);
+    }
+#endif // DEBUG
+
+/*----------------------------- Setting Timeouts -----------------------------*/
+    COMMTIMEOUTS rTimeouts = { 0 }, wTimeouts = { 0 };
+
+    rTimeouts.ReadIntervalTimeout         = 50;
+    rTimeouts.ReadTotalTimeoutConstant    = 50;
+    rTimeouts.ReadTotalTimeoutMultiplier  = 10;
+    rTimeouts.WriteTotalTimeoutConstant   = 50;
+    rTimeouts.WriteTotalTimeoutMultiplier = 10;
+
+    wTimeouts.ReadIntervalTimeout         = 50;
+    wTimeouts.ReadTotalTimeoutConstant    = 50;
+    wTimeouts.ReadTotalTimeoutMultiplier  = 10;
+    wTimeouts.WriteTotalTimeoutConstant   = 50;
+    wTimeouts.WriteTotalTimeoutMultiplier = 10;
+
+    if (
+        SetCommTimeouts(hCommR, &rTimeouts) == FALSE ||
+        SetCommTimeouts(hCommW, &wTimeouts) == FALSE
+    ) {
+        printf("Error >> The SetCommTimeouts() function ended with an error\n");
+        return finish_program(EXIT_FAILURE);
+    }
+#ifdef _DEBUG
+    else printf("Serial Ports Timeout Setting Succeeded\n");
+#endif // _DEBUG
+
+/*--------------------------- Setting Receive Mask ---------------------------*/
+    //Configure Windows to Monitor the serial device for Character Reception
+    if (SetCommMask(hCommR, EV_RXCHAR) == FALSE) {
+        printf("Error >> The SetCommMask() function ended with an error\n");
+        return finish_program(EXIT_FAILURE);
+    }
+#ifdef _DEBUG
+    else printf("Communication mask port setup successful\n");
+#endif // _DEBUG
+
+/*-------------------- Writing a Character to Serial Port --------------------*/
+    if (
+        WriteFile(
+            hCommW,             // Handle to the Serialport
+            lpBuffer,           // Data to be written to the port 
+            dNoOFBytestoWrite,  // No of bytes to write into the port
+            &dNoOfBytesWritten, // No of bytes written to the port
+            NULL
+        ) == TRUE
+    ) {
+        printf("\n%s - Written to %s\n", lpBuffer, wComPortName);
+    }
+    else {
+        printf("\nError >>> %d in Writing to Serial Port\n", GetLastError());
+        return finish_program(EXIT_FAILURE);
+    }
+
+/*------------------------- Setting WaitComm() Event -------------------------*/
+/*----------- Program will Wait here till a Character is received ------------*/
+    printf("\n============== Waiting for Data Reception ===============\n\n");
+    //Wait for the character to be received
+    if (WaitCommEvent(hCommR, &dwEventMask, NULL) == FALSE) {
+        printf("Error >> The WaitCommEvent() function ended with an error\n");
+        return finish_program(EXIT_FAILURE);
+    }
+    else {
+        //Read the RXed data using ReadFile();
```

Comment:
Write the word in full. Don't be misled.

Reply:
Will fix

Reply:
Fixed